wondrous-app/components/RoleModal/RoleModalElement.tsx

```diff
@@ -0,0 +1,211 @@
+import { useEffect, useState } from 'react';
+import { PERMISSIONS } from 'utils/constants';
+import { ActionButton } from 'components/Common/Task/styles';
+import { GET_TOKEN_INFO, GET_NFT_INFO } from 'graphql/queries';
+import ChecklistRow from 'components/CheckList/ChecklistRow';
+import RolePill from 'components/Common/RolePill';
+import apollo from 'services/apollo';
+import NoRolesIcon from 'components/Icons/noRolesIcon';
+import { DiscordIcon } from 'components/Icons/discord';
+import { Tooltip, CircularProgress } from '@mui/material';
+import useGuildXyz from 'services/guildxyz';
+import SuccessRoleModal from 'components/RoleModal/SuccessRoleModal';
+
+import {
+  RequestModalCheckbox,
+  RequestModalCheckPillCombo,
+  RequestModalCloseIcon,
+  RequestModalContainer,
+  RequestModalHelperContainer,
+  RequestModalHelperDiv,
+  RequestModalHorizontalAlign,
+  RequestModalLockedIconOutline,
+  SuccessLockedIconOutline,
+  RequestModalNoRolesContainer,
+  RequestModalNoRolesSubtitle,
+  RequestModalRolesAbilityColumns,
+  RequestModalRolesAbilityContainer,
+  RequestModalRolesSubtitle,
+  RequestModalTitle,
+  RequestModalTitleBar,
+  RequestModalTokenGatingLockBackground,
+  RequestModalTokenGatingSubtitle,
+} from './styles';
+
+const AccessConditionDispaly = (props) => {
+  const { role } = props;
+  const tokenAccessCondition = role?.tokenGatingCondition?.tokenAccessCondition?.[0];
+  const guildAccessCondition = role?.tokenGatingCondition?.guildAccessCondition;
+  const { getGuildById } = useGuildXyz();
+  const [guildRoleInfo, setGuildRoleInfo] = useState(null);
+  const [tokenGatingInfo, setTokenGatingInfo] = useState(null);
+  useEffect(() => {
+    const fetchGuildRole = async () => {
```

Comment:
optional really but can we move the function outside the useEffect and pass the params as arguments